framework/db/BatchQueryResult.php

```diff
@@ -118,7 +118,11 @@ public function next()
             if ($this->query->indexBy !== null) {
                 $this->_key = key($this->_batch);
             } elseif (key($this->_batch) !== null) {
-                $this->_key++;
+                if (null === $this->_key) {
```

Comment:
I think you could put the same code of line 131 here. (Only for consistency)
`$this->_key = $this->_key === null ? 0 : $this->_key + 1;`